src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -20,73 +29,147 @@ const createNGrams = function (words, n) {
   return ngrams
 }
 
-const ngramSizes = [1, 2, 3];
+const cleanKeywords = computed(() => {
+  return cleanDescription(keywordText.value).split(' ').filter(word => word.length > 0 && !filterSet.has(word));
+})
+
+const ngramSizes = computed(() => {
+  const sizes = [];
+  for (let i = 0; i < MAX_VALUE_OF_NGRAM_SIZE; i++) {
+    if (selectedNgramSizes.value[i]) {
+      sizes.push(i + 1)
+    }
+  }
+  return sizes;
+})
 
-const ngramsList = ngramSizes.map(size => {
-  return {
-    size,
-    grams: computed(() => createNGrams(cleanKeywords.value, size))
-  };
+const ngramsList = computed(() => {
+  return ngramSizes.value.map(size => {
+    return {
+      size,
+      grams: computed(() => createNGrams(cleanKeywords.value, size))
+    };
+  });
 });
 </script>
+
 <template>
   <div class="ma-keywords-generator">
     <div class="ma-header">
       <span>Keyword Generator</span>
     </div>
-    <div class="ma-keyword-text-wrapper">
-      <textarea v-model="keywordText" placeholder="enter the text to generate keywords from"
-                class="ma-keyword-textarea"></textarea>
+    <div class="ma-keyword-text-input-wrapper">
+      <MaInput
```

Comment:
we use kebab case for our components like `<ma-input>` please change the related all components